tslint.json

```diff
@@ -7,6 +7,22 @@
     "no-unused-variable": [true, { "ignore-pattern": "^_" }],
     "no-var-requires": false,
     "space-before-function-paren": false,
-    "variable-name": [true, "ban-keywords", "check-format", "allow-pascal-case", "allow-leading-underscore"]
+    "variable-name": [
+      true,
+      "ban-keywords",
+      "check-format",
+      "allow-pascal-case",
+      "allow-leading-underscore"
+    ],
+    "function-name": [
+      true,
+      {
+        "method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]*\\]?$",
+        "private-method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]*\\]?$",
+        "protected-method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]*\\]?$",
+        "static-method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]*\\]?$",
+        "function-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]*\\]?$"
```

Comment:
are these really needed? seems a lot of extra config

Reply:
I'm not sure that they're **_all_** needed. This was based on a solution to the general problem of tslint not properly supporting symbols, taken from here: https://github.com/microsoft/tslint-microsoft-contrib/issues/459. I can add a comment referencing that? Given that tslint is deprecated/unmaintained, I'd lean towards not changing this and instead having a separate effort to migrate from tslint to eslint (more like our setup in Open Elements).